src/components/lib/cos_kernel_api.c

```diff
@@ -640,6 +644,79 @@ cos_sinv(sinvcap_t sinv, word_t arg1, word_t arg2, word_t arg3, word_t arg4)
 	return call_cap_op(sinv, 0, arg1, arg2, arg3, arg4);
 }
 
+/*
+ * TODO:
+ * talk about this naming choice
+ * can't remember what Gabe thought was best choice here
+ */
+void *
```

Comment:
I'd prefer it to be part of a llbooter_lib. 
Just breaks the abstraction in `cos_kernel_api`!! 
IMO `hypervisor` is a misnomer.
Just `llbooter_res_get()` would do it if we make a small library for that with the llbootlib.h defining those enumerations and just this API.

Even better:
Because all these are making sync invs to the booter, why not make llbooter interface with different functions and let the loader make them for us?

Reply:
I agree that this seems out of place in `cos_kernel_api`.  Unfortunately our proxy for "requires low-level access to direct capabilities to resources" is "uses cos_kernel_api".  Perhaps the "right way" to handle this is the work that you're doing (@phanikishoreg) with the resource manager so that the component has access to *no* capabilities other than those given to it by the resource manager.

In the end, this call is a hack.  It is likely a hack that solves a problem with multiple booters are hierarchically arranged (i.e. *not* when we use the resource manager).  Given this, I think that @phanikishoreg is right:  This code should be in some part of the booter's library.

It cannot use an interface as the linker does not look at the booter's API to satisfy sinvs right now, and I don't want to hack that in.